### PR TITLE
examples: add example for ORCA load reporting

### PR DESCRIPTION
**Pull request by dfawley**
cc @temawi 

RELEASE NOTES:
* examples: add example for ORCA load reporting

### CODE REVIEW

File: examples/features/orca/client/main.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+	"google.golang.org/grpc/connectivity"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/orca"
+
+	v3orcapb "github.com/cncf/xds/go/xds/data/orca/v3"
+	pb "google.golang.org/grpc/examples/features/proto/echo"
+)
+
+var addr = flag.String("addr", "localhost:50051", "the address to connect to")
+var test = flag.Bool("test", false, "if set, only 1 RPC is performed before exiting")
```
Comment (easwars) [inline examples/features/orca/client/main.go:40]: What is the use of this flag? Can we get rid of it?

File: examples/features/orca/client/main.go [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+func main() {
+	flag.Parse()
+
+	// Set up a connection to the server.  Configure to use our custom LB
+	// policy which will receive all the ORCA load reports.
+	conn, err := grpc.Dial(*addr,
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+		grpc.WithDefaultServiceConfig(`{"loadBalancingConfig": [{"orca_example":{}}]}`),
+	)
```
Comment (easwars) [inline examples/features/orca/client/main.go:50]: Nit: Even though this is just an example, it would be nice if we don't split up lines this way. We should either have everything on a single line, or define a dial options slice to initialize the dial options.

File: examples/features/orca/client/main.go [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+func (p *picker) Pick(info balancer.PickInfo) (balancer.PickResult, error) {
+	return balancer.PickResult{
+		SubConn: p.sc,
+		Done: func(di balancer.DoneInfo) {
+			fmt.Println("Per-call load report received:", di.ServerLoad.(*v3orcapb.OrcaLoadReport).GetRequestCost())
+		},
+	}, nil
+}
+
+type orcaLis struct{}
```
[the remark this answers is not in the document]
Comment (dfawley) [reply examples/features/orca/client/main.go:149]: Done

**Review by dfawley — COMMENTED** [review]
Thanks for the review!

File: examples/features/orca/README.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+### Per-RPC Metrics
+
+The server is set up to report query cost metrics in its RPC handler.  For
+per-RPC metrics to be reported, the gRPC server must be created with the
+`orca.CallMetricsServerOption()` option, and metrics are set by calling methods
+on the returned `orca.CallMetricRecorder` from
+`orca.CallMetricRecorderFromContext()`.
+
+The client performs one RPC per second.  Per-RPC metrics are available for each
+call via the `Done()` callback returned from the LB policy's picker.
```
Comment (easwars) [inline examples/features/orca/README.md:48]: Most of the other discussion here is about ORCA in general, the two types of reporting metrics, and the API in general. This line `The client performs one RPC per second` seems to be out of place. It seems to suddenly talk about the example.

File: examples/features/orca/client/main.go (hunk 2 above)
Comment (easwars) [reply examples/features/orca/client/main.go:50]: It is part of the style guide: https://g3doc.corp.google.com/go/g3doc/style/guide.md?cl=head#line-length

**Review by easwars — APPROVED** [review]

File: examples/features/orca/README.md (hunk 4 above)
Comment (dfawley) [reply examples/features/orca/README.md:48]: I don't understand.  The whole section is about the example.

"The server is set up to report query cost metrics in its RPC handler. <explanation of how>"
"The client performs one RPC per second. <explanation of how it receives metrics related to those RPCs>"

Can you be more specific about how you would reword this?

Comment (easwars) [reply examples/features/orca/README.md:48]: Actually looks good. Not sure what I was confused about.

File: examples/features/orca/client/main.go (hunk 2 above)
Comment (dfawley) [reply examples/features/orca/client/main.go:50]: Discussed offline & agreed to leave this.
